snuba/query/parser/__init__.py

```diff
@@ -288,6 +289,39 @@ def _expand_aliases(query: Query) -> None:
     query.transform(visitor)
 
 
+DEESCAPER_RE = re.compile(r"^`(.+)`$")
+
+
+def _deescape_aliases(query: Query) -> None:
+    """
+    The legacy query processing does not escape user declared aliases
+    thus aliases like project.name would make the query fail. So Sentry
+    started defining pre-escaped aliases like `project.name` to go
+    around the problem.
+    The AST processing properly escapes aliases thus causing double
+    escaping. We need to de-escape them in the AST query to preserve
+    backward compatibility as log as the legacy query processing is
+    around.
+    """
+
+    def deescape_alias(expr: Expression) -> Expression:
+        if expr.alias is None or not DEESCAPER_RE.match(expr.alias):
+            return expr
+        return replace(expr, alias=expr.alias[1:-1],)
```

Comment:
Would it be clearer just to use the match result value as the alias parameter here?

Reply:
done